profile page contribution counts include unshared/hidden resources

devincowan opened the pull request:
<!--

Please read, and add your text at the bottom of this message.

Thanks for contributing code to HydroShare. In order to maintain code quality and expedite this process, please assist the development team by making sure the following is present in this pull request.

For more information, see https://docs.google.com/document/d/1dzxqlZW5fKNEyQSeKiSFq-SmS-VOPCva95XXkBjPExs

-->

### Pull Request Checklist: 
- [X] Positive Test Case Written by Dev

<!-- Enter steps that a QA engineer, stakeholder, or user documentation writer would follow to test the positive or "successful" case of the functionality your code provides or fixes -->

- [ ] Automated Testing

<!-- Our Jenkins Instance is set up to automatically test every commit from a pull request. Code coverage must not decrease so new functionality or code paths added during a bug fix must have appropriate tests written. Every test must pass, including PEP8 code formatting tests. -->

- [ ] Sufficient User and Developer Documentation

<!-- Please email your positive test case lbrazil@cuahsi.org, who will make the decision regarding user documentation. -->

- [x] Passing Jenkins Build

<!-- Our Jenkins Instance is set up to automatically test every commit from a pull request. Every test must pass, including PEP8 code formatting tests. -->

- [ ] Peer Code review and approval

<!-- This is the process by which a peer developer on the HydroShare team will read the changeset, provide feedback, and ultimately give a formal approval to the code before it passes PR status. -->

### Positive Test Case
1. As a non-admin, navigate to the the Profile->Contributions tab for a user that has unshared+unpublished resources.
2. Make note of the total number of resources shown
3. Click on one of the shared/published resources in the list
4. In the upper-left-hand header on the resource's landing page, click on the Owner link
5. In the Owner info card, you will see that the contributions number matches the total number from the Profile page

#### Next
6. Login as the Owner that has at least one private resource
7. Check the same pages as above and note that the counts still match

***
### OLD Screenshots

#### BEFORE:
![before](https://user-images.githubusercontent.com/17934193/155011999-2f80ed86-7d3e-473a-b9bd-20c547188f9e.png)

#### AFTER:
![after](https://user-images.githubusercontent.com/17934193/155012010-07a242a5-f9b3-4591-98af-2dbdccd7416f.png)

Resolves #4090

hydrocheck commented:
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4504](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4504)

Maurier commented on theme/views.py:
```diff
@@ -80,15 +82,24 @@ def get_context_data(self, **kwargs):
             else:
                 # for anonymous requesting user show only resources that are either public or
                 # discoverable
+                
```
Why is this assignment done twice?

devincowan replied:
That was a mistake, thank you for the catch!

Maurier reviewed (changes requested):
Just one minor comment. However, I don't see the discussion where this was the suggested solution. If this was suggested in a call perhaps we can write a comment on the original thread.

devincowan commented:
@Maurier perhaps I was too liberal here. No discussion. The reason that I added a card instead of just fixing the count badges:

The discrepancy in count is between 
1. author/owner cards in the resource header 
and 
2. resource counts in the user profile contributions tab
The discrepancy arises because non-public resources are included in the vue resource header but not in the profile list.

I don't think we want to reduce the count on the resource header because an owner will see all of the resources in their profile (so if we reduce the count on the resource header, the values would match for non-owners but there would be a discrepancy when owners looked at their own counts).

So the other option is to increase the count on the profile page. That is what I've done here.
I figured it would be confusing if we simply included hidden resources in the "ALL" (total count) on the profile contributions page, without anything showing up in the left list or in the right cards. For example, people would be confused if they only saw 1 contribution card but the "all" total showed 5 (because the owner had 4 hidden resources).

 So I added the hidden resources count to both the left list and the cards.
![image](https://user-images.githubusercontent.com/17934193/155223618-2d6c56d1-87a2-4b7d-88dc-46a8038e73b4.png)

I guess if that is too heavy-handed, I can kick it back for more discussion?

Maurier commented:
Your first option could be to fix the count in the popup to ignore private contributions and change the label to something like 'Public Contributions'. That way we avoid any confusion.

However, your suggested change here also addresses the problem, but creates the concern that it exposes a little bit of information that we didn't before (number of private contributions). This will need input from the team.

devincowan commented:
In conversation with @martinseul I am going to go with your suggestion @Maurier and ignore the private contributions and change the label.

devincowan commented:
@Maurier you can ignore review request on this for now -- I see that it still has some issues...

devincowan commented:
Okay @Maurier the regression I introduced is fixed now...

Maurier reviewed (approved):
Looks good.
